lib/application.js

```diff
@@ -618,6 +618,34 @@ app.listen = function listen() {
   return server.listen.apply(server, arguments);
 };
 
+/**
+ * Creates a router that inherits options from app.settings
+ * and allows them to be overriden by passing in an options
+ * object
+ *
+ * @param {Object} options
+ * @return {router}
+ * @public
+ */
+app.createRouter = function (options) {
+  options = options || {}
+
+  var opts = {
+    strict: this.enabled('case sensitive routing'),
+    caseSensitive: this.enabled('case sensitive routing')
+  }
+
+  if (options.hasOwnProperty('caseSensitive')) {
```

Comment:
Don't use hasOwnProperty to access the key. Just follow the existing patterns so we don't break people who configure with plain objects